### PR TITLE
Message Object On Output Callback 

### PR DESCRIPTION
**Pull request by VictorCoCo**

### CODE REVIEW

File: agt/state.py [hunk 1]
```diff
@@ -44,7 +53,7 @@ def __init__(self, output_callback):
 
         self.memory = {}
 
-    async def say(self, text: str, image_url: str = None):
+    async def say(self, message: Message):
```
Comment (chenb67) [inline agt/state.py]: this is not good it breaks all the existing components

**Review by chenb67 — CHANGES_REQUESTED** [review]

File: agt/state.py (hunk 1 above)
Comment (VictorCoCo) [reply agt/state.py]: Should it be backwards compatible, leaving the text and the image as a parameters and add additional pydantic scheme for ssml and future add-ons? I planned to adjust all the components and started this change as a draft.

File: agt/state.py [hunk 2]
```diff
@@ -44,17 +54,20 @@ def __init__(self, output_callback):
 
         self.memory = {}
 
-    async def say(self, text: str, image_url: str = None):
+    async def say(self, text: str, image_url: str = None, ssml: str = ""):
```
Comment (chenb67) [inline agt/state.py]:
```suggestion
    async def say(self, text: str, image_url: Optional[str] = None, ssml: Optional[str] = None):
```
None so we know when there is no ssml and we can fill it from the text field

File: agt/coco.py [hunk 3]
```diff
@@ -7,7 +7,11 @@
 async def emit_responses(state: ConversationState, component_response: CoCoResponse):
     if hasattr(component_response, "responses") and component_response.responses:
         for resp in component_response.responses:
-            await state.say(text=resp.get("text"), image_url=resp.get("image"))
+            await state.say(
+                text=resp.get("text"),
+                image_url=resp.get("image"),
+                ssml=resp.get("ssml") or "",
```
Comment (chenb67) [inline agt/coco.py]: why empty string

Comment (VictorCoCo) [reply agt/coco.py]: Did not want it to be None at the end of the chain. Can cause exceptions.

**Comment by chenb67** [comment]
there is a bug right now that make it so the responses are accumulated between turns

try it with agt serve and any component

File: agt/server.py [hunk 4]
```diff
@@ -22,14 +22,17 @@ async def default_out_of_context_handler(self, state, user_input=None):
     async def wait_for_out_of_context(self):
         await self.out_of_context_event.wait()
 
-    async def add_response(self, text, *args, **kwargs):
-        self.responses.append(text)
+    async def add_response(self, message: Message, *args, **kwargs):
```
Comment (chenb67) [inline agt/server.py:25]: no need for *args, **kwargs it's all through message now

**Review by chenb67 — APPROVED** [review]

**Pull request merged**